src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -108,6 +110,7 @@ public Task<OidcMetadata> DiscoverOidcMetadataAsync(Uri endpoint, RequestContext
             {
                 AddCommonHeaders(requestContext);
             }
+            AddSsoPolicyHeaders(requestContext);
```

Comment:
This is too deep / affects too many flows. OAuth2Client knows about headers and other HTTP concepts, but not specifically about "SSO policy headers". 

Let's think of a way to set this header only for public client flows.

Reply:
Resolved in another comment